web/app/scripts/stepwise/stepwise-directive.js

```diff
@@ -4,154 +4,302 @@
+                    var svg = d3.select(element[0])
+                        .append('svg')
+                        .attr('width', '100%')
+                        .attr('height', '100%');
 
-                    xRange = isRightToLeft ? [width, 0] : [0, width];
-                    tScale = d3.time.scale().range(xRange);
-                    x = d3.scale.ordinal().rangeBands(xRange, 0.05);
-                    y = d3.scale.linear().range([height, 0]);
+                    //re-render on window resize
+                    $(window).resize('resize.doResize', function() {
+                        render(svg, element, scope.chartData, isRightToLeft);
+                    });
+                    scope.$on('$destroy', function() {
+                        $(window).off('resize.doResize');
+                    });
 
-                    /**
-                     * Watch for changes to chartData and redraw
-                     */
+                    // Watch for changes to chartData and redraw
                     scope.$watch('chartData', function(newVal) {
                         if (newVal) {
-                            var data = formatData(newVal);
-                            updateChart(data, isRightToLeft);
+                            render(svg, element, newVal, isRightToLeft);
                         }
                     });
+                }
 
-                    tooltip = d3.tip().html(function(d) {
-                        return '<strong>' +
-                            $translate.instant('RECORD.WEEK_OF') +
-                            ': </strong>' +
-                            d.dt.toLocaleDateString() +
-                            '</br><strong>' +
-                            $translate.instant('RECORD.EVENT_COUNT') +
-                            ':</strong> <span>' +
-                            d.count +
-                            '</span>';
-                    }).offset([-20,-15]);
+                /**
+                 * Render the chart.
+                 * While updating the chart incrimentally may yield better
+                 * performance, it makes deciphering the state of the chart
+                 * more difficult and results in a bunch of ugly global state
+                 */
+                function render(svg, element, data, isRightToLeft) {
+                    var formattedData = formatData(data);
+                    svg.selectAll('*').remove();
+                    var d3Container = d3.select(element[0])[0][0];
 
-                    var rawSvg = elem.find('svg')[0];
-                    var bufferedWidth = width + margin.left + margin.right;
-                    var bufferedHeight = height + margin.top + margin.bottom;
+                    var dimensions = {
+                        width: d3Container.offsetWidth - 40,
+                        height: d3Container.offsetHeight - 15
+                    };
+                    if (dimensions.width <= 0 || inMapMode) {
```

Comment:
Rather than having this happen implicitly, it would be better to add an explicit configuration setting to the directive. That way it won't behave unexpectedly if this graph is moved to a new location or the display logic is reconfigured on the dashboard or map.